ttnn/CMakeLists.txt

```diff
@@ -15,13 +15,16 @@ target_include_directories(ttnn_lib PUBLIC
     ${CMAKE_CURRENT_SOURCE_DIR}/cpp
     ${CMAKE_SOURCE_DIR}/tt_metal/third_party/fmt
 )
+target_precompile_headers(ttnn_lib PRIVATE
+    ${CMAKE_SOURCE_DIR}/tt_metal/third_party/magic_enum/magic_enum.hpp
```

Comment:
How do you decide what to put into pch?

Reply:
for now, i've just put in third_party header files that are used, so pybinds, etc. since those shouldn't be changing. 

for tt_dnn, i've also include the ones at `tt_dnn/op_library` library, since all the other header files in `op_library` are op specific and only included once or twice at most.